### PR TITLE
Remove jetpack_publicize_post action

### PR DESCRIPTION
**Pull request by enejb**
Since we are using jetpack_published_post instead instead.
- deprecate the action.
- remove listeners
- update tests

This PR is mostly clean up.

#### Testing instructions:
* Publicize should work as expected.

### CODE REVIEW

File: modules/publicize/publicize-jetpack.php [hunk 1]
```diff
@@ -464,7 +464,7 @@ function save_publicized( $post_ID, $post, $update ) {
 				 *
 				 * @param int The post ID
 				 */
-				do_action( 'jetpack_publicize_post', $post->ID );
+				do_action_deprecated( 'jetpack_publicize_post', $post->ID, '4.8.0', 'jetpack_published_post_flags' );
```
Comment (georgestephanis) [inline modules/publicize/publicize-jetpack.php]: For anyone else reviewing this, `do_action_deprecated()` [merged in to Core at 4.6.0](https://github.com/WordPress/WordPress/blob/dd6da701b286579819cd6aa518aa2d7018efd759/wp-includes/plugin.php#L605) and as we support current and current - 1 (currently 4.7 and 4.6) we're fine to use this. 👍

**Comment by lezama** [comment]
🚢

File: modules/publicize/publicize-jetpack.php (hunk 1 above)
Comment (jeherve) [inline modules/publicize/publicize-jetpack.php]: Should we add the hook to our own list of deprecated hooks as well?
https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L5821

**Comment by enejb** [comment]
@jeherve the function you linked to is more intended for functions and methods that we deprecated. Which is a bit different from deprecating an action hook. I think it is fine not to included it there.

**Pull request merged**